File: src/backend/Models/CommonArea.cs
using System.Text.Json.Serialization;

namespace backend.Models
{
    [JsonConverter(typeof(IgnoreIdConverter<CommonArea>))]
    public class CommonArea
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [JsonPropertyName("capacity")]
        public int Capacity { get; set; }

        [JsonPropertyName("rules")]
        public string? Rules { get; set; }

        [JsonPropertyName("created_at")]
        [JsonIgnore]
        public DateTime? CreatedAt { get; set; }

        [JsonPropertyName("updated_at")]
        [JsonIgnore]
        public DateTime? UpdatedAt { get; set; }
    }
}

File: src/frontend-web/src/features/reservations/types/common-area.ts
export type CommonAreaApiRecord = {
  id?: number;
  Id?: number;
  name?: string;
  Name?: string;
  capacity?: number;
  Capacity?: number;
  rules?: string | null;
  Rules?: string | null;
};

export type CommonArea = {
  id: number;
  name: string;
  capacity: number;
  rules: string | null;
};
